pyiron_contrib/atomistic/atomicrex/atomicrex_job.py

```diff
@@ -0,0 +1,127 @@
+"""
+Pyiron interface to atomicrex
+"""
+import xml.etree.ElementTree as ET
+from xml.dom import minidom
+import posixpath
+import os
+import subprocess
+
+from pyiron_base import GenericJob, Settings, PyironFactory
+
+from pyiron_contrib.atomistic.atomicrex.general_input import GeneralARInput
+from pyiron_contrib.atomistic.atomicrex.structure_list import ARStructureList
+from pyiron_contrib.atomistic.atomicrex.potential_factory import ARPotFactory
+from pyiron_contrib.atomistic.atomicrex.output import Output
+
+
+s = Settings()
+
+## Class defined for future addition of other codes
+## Not sure which functionality (if any) can be extracted yet, but a similar pattern is followed in other pyiron modules
+class PotentialFittingBase(GenericJob):
+    def __init__(self, project, job_name):
+        super().__init__(project, job_name)
+
+
+class Atomicrex(PotentialFittingBase):
+    def __init__(self, project, job_name):
+        super().__init__(project, job_name)
+
+        self.__name__ = "atomicrex"
+        self.__version__ = (
+            None
+        )
+        self._executable_activate(enforce=True)
+
+        s.publication_add(self.publication)
+        self.input = GeneralARInput()
+        self.potential = None
+        self.structures = ARStructureList()
+        self.potential_factory = ARPotFactory()
+        ## temprorary set working directory manually before full pyiron integration
+        self._working_directory = None
+        self.output = Output()
+
+    """
+    def to_hdf(self, hdf=None, group_name=None):
```

Comment:
Should call `super().to_hdf(hdf=hdf, group_name=group_name)` and similarly pass those to the other `to_hdf`. Otherwise it should work.